### DISCUSSION TITLE
Debugging the demangler?

### DISCUSSION THREAD

**Discussion by ea** [Vector35/binaryninja-api#4917] [Analysis] [answered]
How would I go about debugging the demangler?
Currently facing an issue where I'm looking at a macOS binary. Function names are properly demangled, but no types are being created as I would expect from "Extract Types From Mangled Names". Is this limited to binaries produced by other compilers ?
Alternatively, where do I look to see what's going wrong? 
This can be tested on any suitable macOS dylib. CoreText for example:
![Screenshot 2024-01-20 at 11 35 38 AM](https://github.com/Vector35/binaryninja-api/assets/65802/db7f2e4c-0fec-4998-ac44-6e72d68ad8da)

Thanks!

**Comment by plafosse** [comment, marked as answer]
This seems like a bug. I'm converting this to an issue.
